File: python/linked-lists/0206-reverse-linked-list.py
from structures import ListNode

def reverseList(head: ListNode) -> ListNode:
    if head is not None:
        cur = head
        prev = None

        while cur.next:
            nxt = cur.next
            cur.next = prev

            prev = cur
            cur = nxt

        cur.next = prev

        head = cur
    return head

File: cpp/linked-lists/0206-reverse-linked-list.cpp
#include <algorithm>
#include <structures.h>

using namespace std;

ListNode* reverseList(ListNode *head){
    if (head == NULL)
        return head;

    ListNode *prev = NULL;
    ListNode *cur = head;

    while (cur != NULL){
        ListNode *temp = cur->next;
        cur->next = prev;
        prev = cur;
        cur = temp;
    }

    head = prev;
    return head;
}